### ISSUE TITLE
Add 'Enable Logs' option for AppSync API 

### ISSUE THREAD

**Issue by hisham** [aws-amplify/amplify-cli#1181]
**Is your feature request related to a problem? Please describe.**

Would like an option for the CLI to enable logging for the AppSync GraphQL api it generates, i.e. turn this on: 

<img width="415" alt="AWS AppSync Console 2019-04-02 11-13-39" src="https://user-images.githubusercontent.com/81853/55422215-61de5100-5538-11e9-8c37-101421422e42.png">

**Describe the solution you'd like**
The CLI to do this and add it to the cloudformation template. 

**Describe alternatives you've considered**
I believe I'll have to modify the cloudformation template myself.

**Comment by hisham** [comment]
Is this something that can be done via a nested stack under the `stacks/` dir, or is the only way is to modify cloudformation-template.json under the `build/` dir which I assume will be overriden on every gql compile?

**Comment by hisham** [comment]
For this to happen, LogConfig needs to be set as part of creating the GraphQLApi. See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-graphqlapi.html#cfn-appsync-graphqlapi-logconfig.

**Comment by mikeparisstuff** [comment]
Agreed. We should add this as part of the add api command. @kaustavghosh06

**Comment by OtterFlip** [comment]
We ran into this as well.  Would be nice to have an easy way to switch it on.  It looks like we and @hisham keep on running into similar stuff.  :)

**Comment by dsmrt** [comment]
+1

Any update on this?

**Comment by nateiler** [comment]
+1  ... Interested as well!

**Comment by joebri** [comment]
+1 Looks like this is still in the 'TODO' column. :(

**Comment by tgjorgoski** [comment]
+1

**Comment by alexkates** [comment]
+1

**Comment by lesmuc** [comment]
+1

**Comment by paulsson** [comment]
+1, it's been almost 1 year since this was opened. Logging is important.
Rule #1 of cloud: if it isn't automated then you're doing it wrong.
I think everyone here can agree with that. ;)
We don't have the luxury of modifying the CloudFormation template since it is re-generated every time there is a API / schema update as part of the transform process and we would lose any modifications made to it to simply enable logging.

This leaves us with the only option of manually enabling Appsync logging, breaking "rule #1 of cloud".
Has anyone found a workaround for this?

**Comment by kaustavghosh06** [comment]
Hey guys, we're looking into this and have added this to our backlog.

**Comment by geoffreyaguero** [comment]
I'm having this issue as well and I believe there's no workaround. I tried to modify the CloudFormation stack but the LogConfig is a property of AWS::AppSync::GraphQLApi and this resource can't be modified since it's created by Amplify during build phase.

**Comment by heitorlessa** [comment]
Similarly, created a new issue to enable Tracing if you'd like to +1 too: https://github.com/aws-amplify/amplify-cli/issues/3835

**Comment by jnathanh** [comment]
+1

**Comment by kylekirkby** [comment]
+1

**Comment by alexkates** [comment]
+1

**Comment by ambriglia** [comment]
As a workaround, my team implemented an amplify backend post build script. We wrote a node app that checks to see whether or not the log config is enabled, if not, it uses the sdk to update it. The AWS docs, found [here](https://docs.aws.amazon.com/appsync/latest/devguide/monitoring.html), mention an IAM role/policy you need to create to enable logging for appsync. We added that into a custom amplify cloudformation template, and then, in the same node app I mentioned above, because it is a post build step, we can reference that role and apply it appropriately. Hopefully this helps others workaround this issue.

**Comment by umbcoppolabottazzi** [comment]
Any update on this?

**Comment by johnjichaowei** [comment]
+1

**Comment by lincetto** [comment]
+1

**Comment by dylan-westbury** [comment]
+1

**Comment by ahtokca** [comment]
Any progress?

**Comment by loganpowell** [comment]
I accidentally deleted the log group created for my AppSync API via Amplify. Is it possible to undo that or attach a new log group? It doesn't seem possible in the console/UI. After a few attempts, it looks like that feature just causes the page to freeze.
